lib/protoparser/zabbixconnector/parser.go

```diff
@@ -0,0 +1,267 @@
+	"github.com/VictoriaMetrics/metrics"
+	"github.com/valyala/fastjson"
+)
+
+const (
+	ADDGROUPS    byte = 0b00000001
+	ADDEMPTYTAGS byte = 0b00000010
+	EMPTYVALUE   byte = 49 // "1"
+)
+
+type Rows struct {
+	Rows []Row
+}
+
+func (rs *Rows) Reset() {
+	for i := range rs.Rows {
+		rs.Rows[i].reset()
+	}
+	rs.Rows = rs.Rows[:0]
+}
+
+func (rs *Rows) Unmarshal(s string, par byte) {
+	rs.Rows = unmarshalRows(rs.Rows[:0], s, par)
+}
+
+// Row is a single Zabbix Connector row
+type Row struct {
+	Tags      []Tag
+	Value     float64
+	Timestamp int64
+}
+
+func (r *Row) reset() {
+	tags := r.Tags
+	for i := range tags {
+		tags[i].reset()
+	}
+	r.Tags = tags[:0]
+
+	r.Value = 0
+	r.Timestamp = 0
+}
+
+func (r *Row) addTag() *Tag {
+	dst := r.Tags
+	if cap(dst) > len(dst) {
+		dst = dst[:len(dst)+1]
+	} else {
+		dst = append(dst, Tag{})
+	}
+	tag := &dst[len(dst)-1]
+	r.Tags = dst
+	return tag
+}
+
+func (r *Row) unmarshal(o *fastjson.Value, par byte) error {
+	r.reset()
+
+	host := o.GetObject("host")
+	if host == nil {
+		return fmt.Errorf("missing `host` object")
+	}
+
+	v := host.Get("host").GetStringBytes()
+	if len(v) == 0 {
+		return fmt.Errorf("missing `host` element in `host` object")
+	}
+	tag := r.addTag()
+	tag.Key = append(tag.Key[:0], []byte("host")...)
+	tag.Value = append(tag.Value[:0], v...)
+
+	v = host.Get("name").GetStringBytes()
+	if len(v) == 0 {
+		return fmt.Errorf("missing `name` element in `host` object")
+	}
+	tag = r.addTag()
+	tag.Key = append(tag.Key[:0], []byte("hostname")...)
+	tag.Value = append(tag.Value[:0], v...)
+
+	v = o.GetStringBytes("name")
+	if len(v) == 0 {
+		return fmt.Errorf("missing `name` element")
+	}
+	tag = r.addTag()
+	tag.Key = append(tag.Key[:0], []byte("__name__")...)
+	tag.Value = append(tag.Value[:0], v...)
+
+	n, err := getFloat64(o, "value")
+	if err != nil {
+		return fmt.Errorf("missing `value` element, %s", err)
+	}
+	r.Value = n
+
+	cl, err := getInt64(o, "clock")
+	if err != nil {
+		return fmt.Errorf("missing `clock` element, %s", err)
+	}
+	ns, err := getInt64(o, "ns")
+	if err != nil {
+		return fmt.Errorf("missing `ns` element, %s", err)
+	}
+	// clock - Number of seconds since Epoch to the moment when value was collected (integer part).
+	// ns - Number of nanoseconds to be added to clock to get a precise value collection time.
+	//
+	// See https://www.zabbix.com/documentation/current/en/manual/appendix/protocols/real_time_export#item-values
+	r.Timestamp = cl*1e3 + ns/1e6
+
+	if (par & ADDGROUPS) != 0 {
+		groups, err := getArray(o, "groups")
+		if err != nil {
+			return fmt.Errorf("missing `groups` element, %s", err)
+		}
+		for _, g := range groups {
+			k := g.GetStringBytes()
+			if len(k) == 0 {
+				continue
+			}
+
+			tag = r.addTag()
+			tag.Key = append(tag.Key[:0], k...)
+			tag.Value = append(tag.Value[:0], EMPTYVALUE)
+		}
+	}
+
+	skipEmptyTags := (par & ADDEMPTYTAGS) == 0
+	item_tags, err := getArray(o, "item_tags")
```

Comment:
```suggestion
	itemTags, err := getArray(o, "item_tags")
```

Please, check out `make check-all` for code-style suggestions.

Reply:
Corrected. Now make check-all does not produce errors or warnings.